mhkit/wave/resource.py

```diff
@@ -684,3 +686,278 @@ def func(kk):
     k = pd.DataFrame(k0, index=f, columns=['k'])
     
     return k
+
+def environmental_contour(x1, x2, dt, period, PCA=None, size_bin=250, 
+                          nb_steps=1000, return_PCA=False):
+    '''    
+    Calculates environmental contours of extreme sea 
+    states using the improved joint probability distributions 
+	with the inverse first-order reliability method (I-FORM) 
+    probability for the desired return period (period). Given the 
+    period of interest a circle of iso-probability is created in the
+    in the principal component analysis (PCA) joint probability 
+    (x1, x2) reference frame.  
+    Using the joint probability value the cumulative distribution function 
+    (CDF)) of the marginal 
+    distribution is used to find the quantile of each component. 
+    Finally, using the improved PCA methodology
+    the component 2 contour lines are calculated from component 1 using 
+    the relationships defined in Exkert-Gallup et. al. 2016.	
+
+    Eckert-Gallup, A. C., Sallaberry, C. J., Dallman, A. R., & 
+    Neary, V. S. (2016). Application of principal component 
+    analysis (PCA) and improved joint probability distributions to 
+    the inverse first-order reliability method (I-FORM) for predicting 
+    extreme sea states. Ocean Engineering, 112, 307-319.
+
+    Parameters
+    ----------
+    x1: array like
+        Component 1 data
+    x2: array like
+        Component 2 data        	
+    dt : float
+        x1 and x2 sample rate (seconds)
+    period : int, float, or numpy array
+        Desired return period (years) for calculation of environmental
+        contour, can be a scalar or a vector.
+    PCA: dict
+	    principal component analysis dictionary from previous function 
+        call. When supplied, the function will skip the PCA calculation 
+        for the passed x1 and x2, and instead use the provided PCA.
+    size_bin : int (optional)
+        Data points in each bin 		
+    nb_steps : int (optional)
+        Discretization of the circle in the normal space used for
+        I-FORM calculation.
+	return_PCA: boolean
+	    Default False, if True will retun the PCA dictionary 
+
```

Comment:
Per previous comment, I think the default should be to return `PCA` if it has been calculated, and not provided. This means I think the `return_PCA` parameter should be removed.